lib/prometheus-metrics.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+class PrometheusMetrics {
+  constructor (circuit) {
+    this.circuit = circuit;
+    // this.client = require('prom-client');
+    this.collect();
+  }
+
+  collect (prefix) {
+    this.collectDefaultMetrics = this.client.collectDefaultMetrics({ prefix: prefix, timeout: 5000 });
+    this.circuit.on('open', () => {
+      this.clear();
+    });
+    this.circuit.on('halfOpen', () => {
+      this.clear();
+    });
+  }
+
+  clear () {
+    clearInterval(this.client.collectDefaultMetrics());
+    this.client.register.clear();
+  }
+
+  getMetrics () {
+    if (this.client.register && this.client.register.metrics) {
+      return this.client.register.metrics();
+    }
+    return "Can't get prometheus metrics.";
+  }
+
+  getClient () {
```

Comment:
This could just be a `get` property on the object. So instead of calling `metrics.getClient()` it becomes `metrics.client` as a read only property.

```suggestion
  get client () {
```